rustls/src/client/hs.rs

```diff
@@ -341,12 +346,14 @@ fn emit_client_hello_for_retry(
         .map(ClientExtension::get_type)
         .collect();
 
+    let s_id = session_id.unwrap_or(SessionID::empty());
```

Comment:
Hmm, this defeats the purpose IMO. We should either go all the way and replace all uses of `empty()` and `is_empty()` with `None` and `is_none()` (in client) or keep it all as is.

(I would also prefer to avoid unrelated formatting changes.)

Reply:
I don't think it does, since it gets the Option handling correct, up to this point. I'll look over the formatting changes.

Reply:
I guess that's true -- "defeats the purpose" is worded too strongly. Would you be up to change it throughout the rest of the client handshake?

Reply:
I think this is the best place to stop for one patch, but I am willing to keep going in another patch today. I'll have to change ClientHelloPayload to keep going, and I think that is best done separately.

Reply:
Fair enough.

Reply:
Can we just rename this `s_id` to `session_id` then? Suddenly giving it a different name doesn't seem helpful -- I think shadowing is more idiomatic in cases like this.

Reply:
Sure thing. Done.